lib/internal/test_runner/test.js

```diff
@@ -76,15 +75,40 @@ const { testNamePatterns, testOnlyFlag } = parseCommandLine();
 let kResistStopPropagation;
 
 function stopTest(timeout, signal) {
+  const deferred = createDeferredPromise();
+
   if (timeout === kDefaultTimeout) {
-    return once(signal, 'abort');
+    signal.addEventListener('abort', deferred.resolve, { __proto__: null, once: true });
+    deferred.cleanup = () => signal.removeEventListener('abort', deferred.resolve);
+    return deferred;
+  }
+
+  const timer = setTimeout(() => {
+    deferred.resolve();
+    signal.removeEventListener('abort', throwOnAbort);
+  }, timeout);
+
+  timer.unref();
+  function throwOnAbort() {
+    deferred.reject(new AbortError(undefined, { cause: signal.reason }));
+    clearTimeout(timer);
   }
```

Comment:
Done this to match the current behavior:
https://github.com/nodejs/node/blob/78f6952751cc422c5cce654049f2d37199297a18/lib/timers/promises.js#L48

but we should align it with just using an abort signal with no timeout - no error being thrown or throw an abort error